File: Source/AlphaTab.Test/Model/TuningParserTest.cs
using AlphaTab.Model;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace AlphaTab.Test.Model
{
    [TestClass]
    public class TuningParserTest
    {
        [TestMethod]
        public void TestStandard()
        {
            var standard = Tuning.GetDefaultTuningFor(6);
            var tuningText = new[] {"E4", "B3", "G3", "D3", "A2", "E2"};

            var tuning = new int[tuningText.Length];
            var tuningText2 = new string[tuningText.Length];
            for (int i = 0; i < tuningText.Length; i++)
            {
                tuning[i] = TuningParser.GetTuningForText(tuningText[i]);
                tuningText2[i] = Tuning.GetTextForTuning(tuning[i], true);
            }

            Assert.AreEqual(string.Join(",", standard.Tunings), string.Join(",", tuning));
            Assert.AreEqual(string.Join(",", tuningText), string.Join(",", tuningText2));
        }
    }
}

File: test/model/TuningParser.test.ts
import { Tuning } from '@src/model/Tuning';
import { ModelUtils } from '@src/model/ModelUtils';

describe('TuningParserTest', () => {
    it('standard', () => {
        let standard: Tuning = Tuning.getDefaultTuningFor(6)!;
        let tuningText: string[] = ['E4', 'B3', 'G3', 'D3', 'A2', 'E2'];
        let tuning = new Array<number>(tuningText.length);
        let tuningText2: string[] = new Array<string>(tuningText.length);
        for (let i: number = 0; i < tuningText.length; i++) {
            tuning[i] = ModelUtils.getTuningForText(tuningText[i]);
            tuningText2[i] = Tuning.getTextForTuning(tuning[i], true);
        }
        expect(tuning.join(',')).toEqual(standard.tunings.join(','));
        expect(tuningText2.join(',')).toEqual(tuningText.join(','));
    });
});
